internal/rbd/replicationcontrollerserver.go

```diff
@@ -237,20 +260,89 @@ func (rs *ReplicationServer) EnableVolumeReplication(ctx context.Context,
 		}
 	}
 
-	if interval != "" {
-		err = rbdVol.addSnapshotScheduling(interval, startTime)
+	return &replication.EnableVolumeReplicationResponse{}, nil
+}
+
+// getDummyImageName returns the csi-vol-dummy+cluster FSID as the image name.
+// each cluster should have a unique dummy image created. choosing the cluster
+// FSID for the same reason.
+func getDummyImageName(conn *util.ClusterConnection) (string, error) {
+	id, err := conn.GetFSID()
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("csi-vol-dummy-%s", id), nil
+}
+
+// getOperationName returns the operation name for the given operation type
+// combined with the pool name.
+func getOperationName(poolName string, optName operation) string {
+	return fmt.Sprintf("%s/%s", poolName, optName)
+}
+
+// createDummyImage creates a dummy image as a workaround for the rbd
+// scheduling problem.
+func createDummyImage(ctx context.Context, rbdVol *rbdVolume) error {
+	optName := getOperationName(rbdVol.Pool, dummyImageCreated)
+	if _, ok := operationLock.Load(optName); !ok {
+		// create a dummy image
+		imgName, err := getdummyImageName(rbdVol.conn)
 		if err != nil {
-			return nil, err
+			return err
 		}
-		log.DebugLog(
-			ctx,
-			"Added scheduling at interval %s, start time %s for volume %s",
-			interval,
-			startTime,
-			rbdVol)
+		dummyVol := rbdVol
+		dummyVol.RbdImageName = imgName
+		err = createImage(ctx, dummyVol, dummyVol.conn.Creds)
+		if err != nil && !strings.Contains(err.Error(), "File exists") {
```

Comment:
@Madhu-1 one quick question here, if the image exist, do we need to return error? rather cant we treat it as success ?

Reply:
Yes its already exists we need to treat it as a success. Example csi deployment restarts etc.